src/clients/c++/examples/CMakeLists.example

```diff
@@ -26,6 +26,8 @@
 
 cmake_minimum_required (VERSION 3.5)
 
+set(CMAKE_CXX_STANDARD 11)
```

Comment:
Unlike nvidia/cuda:10.1-devel-ubuntu-18, nvidia/cuda:10.1-devel-centos7 image contains a version of GCC (4.8.5) which doesn't default to a >=c++11 standard. This makes some files (eg. mutex implementations used by protobuf) to throw an error specifying that the standard of at least c++11 needs to be declared.